Conceptpower+Spring/src/test/java/edu/asu/conceptpower/app/db/DBRequestClientTest.java

```diff
@@ -1,53 +1,58 @@
 package edu.asu.conceptpower.app.db;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-
 import org.junit.Before;
 import org.junit.Test;
+import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.Mockito;
 import org.mockito.MockitoAnnotations;
 
 import com.db4o.ObjectContainer;
+import com.db4o.query.Predicate;
 
+import edu.asu.conceptpower.app.db.CustomMatchers.PredicateReviewRequestMatcher;
+import edu.asu.conceptpower.app.db.CustomMatchers.ReviewRequestMatcher;
 import edu.asu.conceptpower.core.ReviewRequest;
-import edu.asu.conceptpower.core.ReviewStatus;
 
 public class DBRequestClientTest {
     
-    @Mock
+    @InjectMocks
     private DBRequestClient client;
     
     @Mock
     private ObjectContainer objectContainerClient;
     
-    private ReviewRequest reviewRequest = new ReviewRequest();
+    private static final String REVIEW_ID = "REVIEWf78d0a6e-a187-43df-b3be-3f22c040b5a1";
+    private static final String CONCEPT_ID = "WID-10126926-N-05-Einstein";
     
     @Before
     public void init() {
         MockitoAnnotations.initMocks(this);
+    }
+    
+    @Test
+    public void test_getReview_success() {        
+        Predicate<ReviewRequest> request = new Predicate<ReviewRequest>() {
+            private static final long serialVersionUID = 6495914730735826451L;
+
+            @Override
+            public boolean match(ReviewRequest review) {
+                return review.getId().equals(REVIEW_ID);
+            }
+            
+        };
         
-        objectContainerClient = Mockito.mock(ObjectContainer.class);
-        
-        reviewRequest.setConceptId("WID-10126926-N-05-Einstein");
-        reviewRequest.setRequest("Review the Spelling");
-        reviewRequest.setRequester("user2");
-        reviewRequest.setResolver("admin");
-        reviewRequest.setStatus(ReviewStatus.OPENED);
 
-        Mockito.when(client.getReviewRequestForConcept(reviewRequest.getConceptId())).thenReturn(reviewRequest);
+        client.getReview(REVIEW_ID);
+        Mockito.verify(objectContainerClient).query(Mockito.argThat(new PredicateReviewRequestMatcher(request, REVIEW_ID)));
```

Comment:
why do you verify that the objectContainerClient is called instead of making sure the returned review request object is correct?

Reply:
Both query() and queryByExample() are expecting the return object in ObjectSet<Object> type. I couldn't type cast a List of reviewRequests or plain objects only. Thats why I only verified whether the specific function is called with the expected parameter.

Reply:
since objectset is an interface, why don't you let mockito mock it for you or simply provide a dummy implementation?